pkg/live/planner/cluster.go

```diff
@@ -0,0 +1,289 @@
+
+	eventCh := r.applier.Run(ctx, inv, objects, apply.ApplierOptions{
+		DryRunStrategy:    common.DryRunServer,
+		ServerSideOptions: o.ServerSideOptions,
+	})
+
+	var actions []planv1alpha1.Action
+	var err error
+	for e := range eventCh {
+		if e.Type == event.InitType {
+			// This event includes all resources that will be applied, pruned or deleted, so
+			// we make sure we fetch all the resources from the cluster.
+			// TODO: See if we can update the actuation library to provide the pre-actuation
+			// versions of the resources as part of the regular run. This solution is not great
+			// as fetching all resources will take time.
+			a, err := r.fetchResources(ctx, e)
+			if err != nil {
+				return nil, err
+			}
+			actions = a
+		}
+		if e.Type == event.ErrorType {
+			// Update the err variable here, but wait for the channel to close
+			// before we return from the function.
+			// Since ErrorEvents are considered fatal, there should only be sent
+			// and it will be followed by the channel being closed.
+			err = e.ErrorEvent.Err
+		}
+		// For the Apply, Prune and Delete event types, we just capture the result
+		// of the dry-run operation for the specific resource.
+		switch e.Type {
+		case event.ApplyType:
+			id := e.ApplyEvent.Identifier
+			index := indexForIdentifier(id, actions)
+			a := actions[index]
+			actions[index] = handleApplyEvent(e, a)
+		case event.PruneType:
+			id := e.PruneEvent.Identifier
+			index := indexForIdentifier(id, actions)
+			a := actions[index]
+			actions[index] = handlePruneEvent(e, a)
+		case event.DeleteType:
+			id := e.DeleteEvent.Identifier
+			index := indexForIdentifier(id, actions)
+			a := actions[index]
+			actions[index] = handleDeleteEvent(e, a)
+		}
+	}
+	return actions, err
+}
+
+func handleApplyEvent(e event.Event, a planv1alpha1.Action) planv1alpha1.Action {
+	if e.ApplyEvent.Error != nil {
+		a.Type = planv1alpha1.Error
+		a.Error = e.ApplyEvent.Error.Error()
+	} else {
+		switch e.ApplyEvent.Operation {
+		case event.Unchanged:
+			a.Type = planv1alpha1.Skip
+		case event.ServersideApplied:
+			a.After = e.ApplyEvent.Resource
+			if a.Before != nil {
+				// TODO: Unclear if we should diff the full resources here. It doesn't work
+				// well with client-side apply as the managedFields property shows up as
+				// changes. It also means there is a race with controllers that might change
+				// the status of resources.
+				if reflect.DeepEqual(a.Before, a.After) {
+					a.Type = planv1alpha1.Unchanged
+				} else {
+					a.Type = planv1alpha1.Update
+				}
+			} else {
+				a.Type = planv1alpha1.Create
+			}
+		}
+	}
+	return a
+}
+
+func handlePruneEvent(e event.Event, a planv1alpha1.Action) planv1alpha1.Action {
+	if e.PruneEvent.Error != nil {
+		a.Type = planv1alpha1.Error
+		a.Error = e.PruneEvent.Error.Error()
+	} else {
+		switch e.PruneEvent.Operation {
+		case event.Pruned:
+			a.Type = planv1alpha1.Delete
+		case event.PruneSkipped:
+			a.Type = planv1alpha1.Skip
```

Comment:
Q: Why might prune get skipped?  A comment here or TODO to figure out why things are skipped would be great

Reply:
Added a comment here. Will add more information about why resources might be skipped in a follow-up PR.